test/integration/targets/azure_rm_virtualmachine_scaleset/tasks/main.yml

```diff
@@ -248,6 +285,54 @@
       - 'results.ansible_facts.azure_vmss.properties.virtualMachineProfile.networkProfile.networkInterfaceConfigurations.0.properties.enableAcceleratedNetworking == true'
       - 'results.ansible_facts.azure_vmss.properties.virtualMachineProfile.networkProfile.networkInterfaceConfigurations.0.properties.networkSecurityGroup != {}'
 
+- name: Create VMSS with security group in same resource group, with accelerated networking.
+  azure_rm_virtualmachine_scaleset:
+    resource_group: "{{ resource_group }}"
+    name: testVMSS{{ rpfx }}2
+    vm_size: Standard_D3_v2
+    capacity: 1
+    virtual_network_name: testVnet
+    subnet_name: testSubnet
+    admin_username: testuser
+    ssh_password_enabled: true
+    admin_password: "Password1234!"
+    image:
+      name: testimagea
+      resource_group: "{{ resource_group }}"
+    upgrade_policy: Manual
+    security_group: testNetworkSecurityGroup
+    enable_accelerated_networking: yes
+  register: results
+
+- name: Assert that nothing has changed
+  assert:
+    that:
+      - not results.changed
+
+- name: Create VMSS with security group in same resource group, with accelerated networking.
```

Comment:
minor thing, pls update the test name to mention that image reference was updated also.